internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -140,9 +142,33 @@ func (ctrl *ControlPlaneStaticPodController) Run(ctx context.Context, r controll
 				return fmt.Errorf("error getting control plane config: %w", err)
 			}
 
-			if err = pod.f(ctx, r, logger, res.(*config.K8sControlPlane), secretsVersion); err != nil {
+			var podID string
+
+			if podID, err = pod.f(ctx, r, logger, res.(*config.K8sControlPlane), secretsVersion); err != nil {
```

Comment:
This feels a bit too implicit to me:  if the podID is not returned, then it should be deleted?  Could we instead get a more explicit signal?  What happens if one of these `manageXXX()` functions gets expanded and has the ability to return errors which do _not_ indicate that the pod should be destroyed?  That would leads to self-destruction due to a transitent error.

I wonder if, perhaps, we have sufficient data here to be able to just preload these controlplane component structs with the administrative status of them from their configs.

Reply:
if we can't build the pod spec, I think we should destroy the running one, as otherwise the intention (configuration) gets out of sync with the running state.

Reply:
Fair point; I'm still just concerned that this leads to the potential of externalities being responsible for taking down the controlplane in ways which are not easily diagnosable.  I'm not going to stand on this objection, though.
Perhaps we can moderate it by having sufficient visibility into any error encountered (log when a controlplane pod is being removed and the reason for it)

Reply:
Eh, we already do, for the most part.  If we return empty _and_ a nil error, that's an error on us (upstream).